src/validators/keystores/hashi_vault.py

```diff
@@ -38,52 +44,57 @@ def from_settings(cls) -> 'HashiVaultConfiguration':
             token=settings.hashi_vault_token,
             url=settings.hashi_vault_url,
             engine_name=settings.hashi_vault_engine_name,
-            key_paths=settings.hashi_vault_key_paths,
+            key_paths=settings.hashi_vault_key_paths or [],
+            key_prefixes=settings.hashi_vault_key_prefixes or [],
             parallelism=settings.hashi_vault_parallelism,
         )
 
-    def secret_url(self, key_path: str) -> str:
+    def secret_url(self, key_path: str, location: str = 'data') -> str:
         return urllib.parse.urljoin(
             self.url,
-            f'/v1/{self.engine_name}/data/{key_path}',
+            f'/v1/{self.engine_name}/{location}/{key_path}',
         )
 
+    def prefix_url(self, keys_prefix: str) -> str:
+        """An URL for Vault secrets engine location that holds prefixes for keys."""
+        keys_prefix = keys_prefix.strip('/')
+        # URL is used for listing, so it lists metadata
+        return self.secret_url(keys_prefix, location='metadata')
+
+
+@dataclass
+class HashiVaultKeysLoader(metaclass=abc.ABCMeta):
+    config: HashiVaultConfiguration
+    input_iter: Iterator[str]
+
+    def session(self) -> AsyncContextManager:
```

Comment:
```suggestion
    def session(self) -> ClientSession:
```

Reply:
Fixed in https://github.com/stakewise/v3-operator/pull/439/commits/7def6db7a02737dc41f3c33482fcf03b12d76521